contrib/coredns/policy/dns.go

```diff
@@ -180,3 +213,59 @@ func ip2rr(ip net.IP, name string, class uint16) dns.RR {
 	}
 	return nil
 }
+
+// find record for typ if in rrCodeFormat
+func findRecord(dst string, typ uint16) (string, bool) {
+	rrCodeFormat := false
+	for _, each := range strings.Split(dst, rrCodeDelimiter) {
+		for _, rrCodeTyp := range rrCodeTypes {
+			prefix := dns.TypeToString[rrCodeTyp] + rrCodePrefixDelimiter
```

Comment:
undesired concatenation/allocation. You can do
`if strings.HasPrefix(each, dns.TypeToString[rrCodeTyp]) {`
and then check for upcoming delimiter